### PR TITLE
Convert API service, last fm importer and tests to typescript

### PR DESCRIPTION
**Pull request by paramsingh**
# Description

We're moving over to Typescript for our frontend code. This pull request migrates the lastfm importer and all related files (tests and the api service) to typescript.

We also do some linting fixes here.

# Problem

The LB frontend code is very hard to read or modify because it's complex and javascript provides no guarantee of what is available where etc.

# Solution

Use typescript instead, which helps us have types on react components and functions and catch errors at compile time.

The linter has also caught a few actual bugs too. 

# Action

Merge and deploy to beta, monitor for a few days and then deploy to production.

### CODE REVIEW

File: listenbrainz/webserver/static/js/src/APIService.ts [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+            Authorization: `Token ${userToken}`,
+            "Content-Type": "application/json;charset=UTF-8",
+          },
+          body: JSON.stringify(struct),
+        });
+
+        // we skip listens if we get an error code that's not a rate limit
+        if (response.status === 429) {
+          // Rate limit error, this should never happen, but if it does, try again in 3 seconds.
+          setTimeout(
```
Comment (paramsingh) [inline listenbrainz/webserver/static/js/src/APIService.ts:117]: This recursion is risky and should be rewritten.

File: listenbrainz/webserver/static/js/src/APIService.ts [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+            3000
+          );
+        }
+        return response; // Return response so that caller can handle appropriately
+      } catch {
+        // Retry if there is an network error
+        setTimeout(
+          () => this.submitListens(userToken, listenType, payload),
+          3000
+        );
```
Comment (ishaanshah) [inline listenbrainz/webserver/static/js/src/APIService.ts:127]: Something should be returned otherwise extra calls would be made below

**Comment by paramsingh** [comment]
@ishaanshah please take a look, this is ready for review

File: listenbrainz/webserver/static/js/src/APIService.test.ts [hunk 3]
```diff
@@ -81,7 +71,6 @@ describe("submitListens", () => {
     });
 
     await apiService.submitListens("foobar", "import", "foobar");
-    expect(console.warn).toHaveBeenCalledWith("Got 404 error, skipping");
```
[the remark this answers is not in the document]
Comment (paramsingh) [reply listenbrainz/webserver/static/js/src/APIService.test.ts:84]: Added an assert that setTimeout is not called here.

File: listenbrainz/webserver/static/js/src/APIService.ts (hunk 2 above)
Comment (paramsingh) [reply listenbrainz/webserver/static/js/src/APIService.ts:127]: Should be fixed as a part of this ticket: https://tickets.metabrainz.org/browse/LB-516

File: listenbrainz/webserver/static/js/src/APIService.ts (hunk 1 above)
Comment (paramsingh) [reply listenbrainz/webserver/static/js/src/APIService.ts:116]: https://tickets.metabrainz.org/browse/LB-516

**Comment by paramsingh** [comment]
Self approving and merging, i've addressed @ishaanshah's comments. I want to deploy this to beta now. If there are more comments I'll fix them in subsequent pull requests.

**Pull request merged**